compiler/rustc_query_system/src/query/job.rs

```diff
@@ -591,10 +591,14 @@ pub(crate) fn report_cycle<'a>(
         err.span_note(span, &format!("...which requires {}...", query.description));
     }
 
-    err.note(&format!(
-        "...which again requires {}, completing the cycle",
-        stack[0].query.description
-    ));
+    if stack.len() == 1 {
+        err.note(&format!("...which immediately again requires {}", stack[0].query.description));
```

Comment:
I'm not sure I'm a fan of the new wording. Something about "immediately again" sounds weird to me, too... "lyrical", but I'm ESL so it might be fine.

Reply:
It sounds a bit off to me as well. I thought of doing `...which immediately requires {} again` or `...which immediately requires again {}`, but those sounded weirder to me. Do you have an idea for a better phrasing?

Reply:
I think the first of the options you mention is the most grammatically correct.